cg/services/bcl_convert_metrics_service/parser.py

```diff
@@ -112,6 +112,20 @@ def get_q30_bases_percent_for_sample_in_lane(self, sample_internal_id: str, lane
         )
         return self.calculate_mean_read_pair_q30_bases_percent(metrics=metrics)
 
+    def get_yield_for_sample_in_lane(self, sample_internal_id: str, lane: int) -> float:
+        """Return the yield for a sample and lane."""
+        metrics: list[SequencingQualityMetrics] = self.get_read_pair_metrics_for_sample_and_lane(
+            sample_internal_id=sample_internal_id, lane=lane
+        )
+        return self.calculate_mean_yield(metrics=metrics)
+
+    def get_yield_q30_for_sample_in_lane(self, sample_internal_id: str, lane: int) -> float:
+        """Return the yield Q30 for a sample and lane."""
+        metrics: list[SequencingQualityMetrics] = self.get_read_pair_metrics_for_sample_and_lane(
+            sample_internal_id=sample_internal_id, lane=lane
+        )
+        return self.calculate_mean_yield_q30(metrics=metrics)
```

Comment:
I find these a bit confusing. We specify both sample_internal_id and lane. Why would we get a list of `SequencingQualityMetrics` back, and why would we then calculate the mean of one thing?

Reply:
Is it because there is one entry per read direction?

Reply:
Read 1 and Read2 indeed